generated/nidcpower/nidcpower/_library_interpreter.py

```diff
@@ -0,0 +1,707 @@
+# -*- coding: utf-8 -*-
+# This file was generated
+
+import array
+import ctypes
+import hightime  # noqa: F401
+import nidcpower._converters as _converters  # noqa: F401
+import nidcpower._library_singleton as _library_singleton
+import nidcpower._visatype as _visatype
+import nidcpower.enums as enums  # noqa: F401
+import nidcpower.errors as errors
+
+import nidcpower.lcr_load_compensation_spot as lcr_load_compensation_spot  # noqa: F401
+
+import nidcpower.lcr_measurement as lcr_measurement  # noqa: F401
+
+
+# Helper functions for creating ctypes needed for calling into the driver DLL
+def get_ctypes_pointer_for_buffer(value=None, library_type=None, size=None):
+    if isinstance(value, array.array):
+        assert library_type is not None, 'library_type is required for array.array'
+        addr, _ = value.buffer_info()
+        return ctypes.cast(addr, ctypes.POINTER(library_type))
+    elif str(type(value)).find("'numpy.ndarray'") != -1:
+        import numpy
+        return numpy.ctypeslib.as_ctypes(value)
+    elif isinstance(value, bytes):
+        return ctypes.cast(value, ctypes.POINTER(library_type))
+    elif isinstance(value, list):
+        assert library_type is not None, 'library_type is required for list'
+        return (library_type * len(value))(*value)
+    else:
+        if library_type is not None and size is not None:
+            return (library_type * size)()
+        else:
+            return None
+
+
+def get_ctypes_and_array(value, array_type):
+    if value is not None:
+        if isinstance(value, array.array):
+            value_array = value
+        else:
+            value_array = array.array(array_type, value)
+    else:
+        value_array = None
+
+    return value_array
+
+
+class LibraryInterpreter(object):
+    '''Library C<->Python interpreter.
+
+    This class handles ctypes conversions, string encoding, and raising exceptions if the library
+    function returns an error.
+    '''
+
+    def __init__(self, encoding):
+        self._encoding = encoding
+        self._library = _library_singleton.get()
```

Comment:
Consider making library a `__init__` parameter.

Will clean up unit tests a bit. But more importantly it makes sense to me to have this integration policy be dictated by higher layers of the stack.

Reply:
I may revisit this after gRPC, if it still makes sense. But for now it seems like we'll want the library-specific layers to have that knowledge.